src/Dropdown.tsx

```diff
@@ -0,0 +1,100 @@
+import React, { useMemo, useState } from 'react';
+import { Person } from './types/Person';
+import cn from 'classnames';
+import debounce from 'lodash.debounce';
+
+type Props = {
+  delay: number;
+  people: Person[];
+  onSelected: (person: Person | null) => void;
+};
+
+export const Dropdown: React.FC<Props> = ({ delay, people, onSelected }) => {
+  const [inputValue, setInputValue] = useState('');
+  const [visible, setVisible] = useState(false);
+
+  const debounceName = useMemo(
+    () => debounce(() => setVisible(true), delay),
+    [delay],
+  );
+
+  const filteredPeople = useMemo(() => {
+    return people.filter(person => {
+      const fixedText = inputValue.toLocaleLowerCase().trim();
+
+      return person.name.toLowerCase().includes(fixedText);
+    });
+  }, [people, inputValue]);
+
+  const handleSelectPerson = (person: Person) => {
+    onSelected(person);
+    setInputValue(person.name);
+    setVisible(false);
+  };
+
+  const handleInputValue = (event: React.ChangeEvent<HTMLInputElement>) => {
+    debounceName();
+    setInputValue(event.target.value);
+    onSelected(null);
+    setVisible(false);
+  };
+
+  return (
+    <>
+      <div className={cn('dropdown', { 'is-active': visible })}>
+        <div className="dropdown-trigger">
+          <input
+            type="text"
+            placeholder="Enter a part of the name"
+            className="input"
+            data-cy="search-input"
+            value={inputValue}
+            onChange={handleInputValue}
+            onFocus={() => setVisible(true)}
+          />
+        </div>
+
+        <div className="dropdown-menu" role="menu" data-cy="suggestions-list">
+          {filteredPeople.length > 0 && (
+            <div className="dropdown-content">
+              {visible &&
+                filteredPeople.map(person => (
+                  <div
+                    key={person.slug}
+                    className="dropdown-item"
+                    data-cy="suggestion-item"
+                    onClick={() => handleSelectPerson(person)}
+                  >
+                    <p
+                      className={cn({
+                        'has-text-link': person.sex === 'm',
+                        'has-text-danger': person.sex === 'f',
+                      })}
+                    >
+                      {person.name}
+                    </p>
+                  </div>
+                ))}
+            </div>
+          )}
+        </div>
+      </div>
+
+      {filteredPeople.length === 0 && (
```

Comment:
Well, if you still prefer this way to check :) 

But I suggest to stick to `!!filteredPeople.length `

Reply:
seems that `!filteredPeople.length` works better ;)

Reply:
Yep, was looking on another condition you have :) 
Basically, there are two checks in your code: 
1. To show items when there are filteredPeople
2. Show a message when there are no filteredPeople

For the first one you have `filteredPeople .length > 0` => we can shorten it to `!!filteredPeople.length` => which basically means 'give me a boolean analog of this value' , so whenever it's > 0 (!!0 === false) we will display items 

And for the second one we just want to check that there is no value(length === 0), in this case it's just 
`! filteredPeople.length` => which means 'if there is NO value for the array length', this will evaluate to true only when filteredPeople === []